contact_list_controller.py

```diff
@@ -0,0 +1,47 @@
+import json
+
+
+class Contact:
+    def __init__(self, name, phone):
+        self.name = name
+        self.phone = phone
+
+
+class ContactList:
+    def __init__(self):
+        self.contacts = []
+
+        # load the json when object is created
+
+    def add_contact(self, contact: object) -> None:
+        self.contacts.append(contact)
+
+    def save_contact(self, file_name: str) -> None:
+        with open(file_name, "w") as file:
+            json.dump([vars(contact) for contact in self.contacts], file)
+
+    def search_contact(self, name: str) -> object:
+        for contact in self.contacts:
+            if contact.name == name:
+                return contact
+
+
+contacts = ContactList()
+
+
+def add_record():
+    name = input("Enter name: ")
```

Comment:
@rajan10 , create another function name take_inputs and externalize the code that taking input from console in add_record